src/templates/k8soperator/Content/Operator.cs

```diff
@@ -0,0 +1,138 @@
+﻿using __ProjectName__.Controller;
+using __ProjectName__.Crd;
+using __ProjectName__.Diagnostics;
+using k8s;
+using Microsoft.Extensions.Hosting;
+using System;
+using System.Threading;
+using System.Threading.Channels;
+using System.Threading.Tasks;
+
+namespace __ProjectName__
+{
+    public class Operator : IHostedService
+    {
+        private Watcher<{{crdname}}Resource> _watcher;
+
+        private readonly Channel<WatchData> _channel;
+
+        private readonly CancellationTokenSource _stoppingCts = new CancellationTokenSource();
+        private readonly IKubernetes _kubernetesClient;
+        private readonly I__ProjectName__Controller _controller;
+        private readonly OperatorDiagnostics _diagnostics;
+
+        public Operator(
+            IKubernetes kubernetesClient,
+            I__ProjectName__Controller controller,
+            OperatorDiagnostics diagnostics)
+        {
+            _kubernetesClient = kubernetesClient ?? throw new ArgumentNullException(nameof(kubernetesClient));
+            _controller = controller ?? throw new ArgumentNullException(nameof(controller));
+            _diagnostics = diagnostics ?? throw new ArgumentNullException(nameof(diagnostics));
+
+            _channel = Channel.CreateUnbounded<WatchData>(new UnboundedChannelOptions()
+            {
+                SingleReader = true,
+                SingleWriter = true
+            });
+        }
+
+        public Task StartAsync(CancellationToken cancellationToken)
+        {
+            _diagnostics.OperatorStarting();
+
+            //starting the channel reader and the watcher, we are not awaiting the watcher
+            // because this is on wait state meanwhile not crd objects are present/created 
+            //on K8S and the hosted initialization can't be continue
+
+            Task.Run(WatchListener);
+            _ = StartWatcher(_stoppingCts.Token);

```

Comment:
Why isn't this awaited?

Reply:
Because the call to https://github.com/Xabaril/lucecu/blob/365cbcc5d4f0f694797c1b82bf7b8bccac7bfe07/src/templates/k8soperator/Content/Operator.cs#L74

Is not finished meanwhile K8S cluster does not have any crd object and the hosted service can't continue. With the current aproach, hosted service can continue even when not crd objects are on k8s and we can do a gracefully stop.
Make sense?